integrations/micronaut/cdi/src/main/java/io/helidon/integrations/micronaut/cdi/MicronautCdiExtension.java

```diff
@@ -0,0 +1,405 @@
+
+                    BeanConfigurator<Object> newBean = event.addBean()
+                            .addType(beanType)
+                            .id("micronaut-" + ref.getBeanDefinitionName())
+                            .scope(Dependent.class)
+                            .produceWith(instance -> micronautContext.get().getBean(beanType, composite));
+
+                    for (Annotation annotation : synthesized) {
+                        newBean.addQualifier(annotation);
+                    }
+
+                }
+            }
+        }
+        unprocessedBeans.clear();
+    }
+
+    /**
+     * Start Micronaut application context and make it available for other extensions.
+     *
+     * @param event CDI event
+     */
+    void startContext(@Observes @Priority(PLATFORM_BEFORE) @Initialized(ApplicationScoped.class) Object event) {
+        ApplicationContext context = ApplicationContext.builder()
+                .propertySources(createMicronautPropertySource())
+                .build();
+
+        context.start();
+
+        micronautContext.set(context);
+    }
+
+    /**
+     * Stop Micronaut application context.
+     * @param event CDI event
+     */
+    void stopContext(@Observes @Priority(PLATFORM_AFTER) @BeforeDestroyed(ApplicationScoped.class) Object event) {
+        ApplicationContext context = micronautContext.get();
+        // if startup failed, context is null
+        if (context != null) {
+            context.close();
+        }
+    }
+
+    /**
+     * Use MicroProfile Config as a config source for Micronaut.
+     *
+     * @return Micronaut property source from MP Config
+     */
+    private PropertySource createMicronautPropertySource() {
+        Config config = org.eclipse.microprofile.config.ConfigProvider.getConfig();
+
+        return new PropertySource() {
+            @Override
+            public String getName() {
+                return "MicroProfile-Config";
+            }
+
+            @Override
+            public Object get(String key) {
+                return config.getOptionalValue(key, String.class).orElse(null);
+            }
+
+            @Override
+            public Iterator<String> iterator() {
+                List<String> names = new LinkedList<>();
+                config.getPropertyNames()
+                        .forEach(names::add);
+
+                return names.iterator();
+            }
+        };
+    }
+
+    /**
+     * Load Micronaut bean definition using service loader, to have full knowledge of the landscape.
+     */
+    @SuppressWarnings("rawtypes")
+    private void loadMicronautBeanDefinitions() {
+        // we now need to load all Micronaut beans so other extensions can inject them
+        List<ServiceDefinition<BeanDefinitionReference>> list = new ArrayList<>(200);
+
+        SoftServiceLoader.load(BeanDefinitionReference.class)
+                .forEach(list::add);
+
+        list.stream()
```

Comment:
Minor, but this would preform better with `parallelStream()` and then `collect()` then adding all the beans to `beanDefinitions`